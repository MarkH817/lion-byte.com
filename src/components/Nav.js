import React from 'react'
import { Link } from 'gatsby'
import styled from 'styled-components'

const Menu = styled.nav`
  font-family: ${props => props.theme.headerFont};
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-around;

  a {
    font-weight: bold;
    display: inline-block;
    padding: 0 0.75ch;
    text-align: center;
    text-decoration: none;
  }
`

export const Nav = () => (
  <Menu>
    <Link to='/'>Home</Link>
    <Link to='/about'>About</Link>
    <Link to='/blog'>Blog</Link>
  </Menu>
)

export default Nav
