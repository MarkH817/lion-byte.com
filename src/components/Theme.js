import React from 'react'
import { ThemeProvider, createGlobalStyle } from 'styled-components'
import '@reach/skip-nav/styles.css'
import 'prismjs/themes/prism-okaidia.css'
import 'typeface-nunito'
import 'typeface-open-sans'

const theme = {
  headerFont: `'Nunito', Arial, Helvetica, sans-serif`,
  textFont: `'Open Sans', Arial, Helvetica, sans-serif`,
  blue: '#28579d',
  red: '#e73212',
  black: '#2e2e31',
  grey: '#3a3a3a',
  lightGrey: '#e1e1e1',
  offWhite: '#ededed',
  maxWidth: '750px',
  bs: '0 12px 24px 0 rgba(0, 0, 0, 0.09)'
}

const GlobalStyles = createGlobalStyle`
  * {
    box-sizing: border-box;

    &:focus {
      outline: 0.1rem solid ${props => props.theme.red};
    }

    &::selection {
      background-color: ${props => props.theme.red};
      color: #ffffff;
    }
  }

  body {
    font-family: ${props => props.theme.textFont};
    line-height: 1.3;
    margin: 0;
    padding: 0;
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    font-family: ${props => props.theme.headerFont};
  }

  a {
    color: ${props => props.theme.blue};

    &:active {
      color: ${props => props.theme.red};
    }
  }

  p {
    margin-bottom: 2em;
  }

  [data-reach-skip-link] {
    color: cornflowerblue;
    text-decoration: none;

    &:focus {
      outline-color: currentColor;
    }
  }

  @keyframes fadein {
    0% {
      opacity: 0;
    }

    100% {
      opacity: 1;
    }
  }
`

export const Theme = props => {
  const { children } = props

  return (
    <ThemeProvider theme={theme}>
      <React.Fragment>
        <GlobalStyles />

        {children}
      </React.Fragment>
    </ThemeProvider>
  )
}

export default Theme
