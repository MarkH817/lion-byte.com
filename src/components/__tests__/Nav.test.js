import React from 'react'
import renderer from 'react-test-renderer'

import Theme from '../Theme'
import Nav from '../Nav'

describe('Nav', () => {
  test('Nav renders correctly', () => {
    const tree = renderer
      .create(
        <Theme>
          <Nav />
        </Theme>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
