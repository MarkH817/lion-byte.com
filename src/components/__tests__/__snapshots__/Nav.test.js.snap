// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Nav renders correctly 1`] = `
.c0 {
  font-family: 'Nunito',Arial,Helvetica,sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c1 {
  -webkit-transition: -webkit-transform 0.08s ease-in-out;
  -webkit-transition: transform 0.08s ease-in-out;
  transition: transform 0.08s ease-in-out;
}

.c1:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
}

.c1 a {
  color: #28579d;
  font-weight: bold;
  display: inline-block;
  padding: 0 0.75ch;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media screen and (max-width:450px) {
  .c1 a {
    padding: 0 0.75ch;
  }
}

<nav
  className="c0"
>
  <section
    className="c1"
  >
    <a
      href="/"
    >
      Home
    </a>
  </section>
  <section
    className="c1"
  >
    <a
      href="/about"
    >
      About
    </a>
  </section>
  <section
    className="c1"
  >
    <a
      href="/blog"
    >
      Blog
    </a>
  </section>
  <section
    className="c1"
  >
    <a
      href="/contact"
    >
      Contact
    </a>
  </section>
</nav>
`;
